modules/caddyhttp/reverseproxy/selectionpolicies.go

```diff
@@ -139,14 +139,17 @@ func (r RandomChoiceSelection) Select(pool UpstreamPool, _ *http.Request) *Upstr
 	if k > len(pool) {
 		k = len(pool)
 	}
-	choices := make([]*Upstream, k)
+	// Slice of length 0 and capacity k (if we use an array and have some unavailable
+	// upstream, we could have some empty element which may lead
+	// to an index out of range runtime error)
```

Comment:
I think this logic is pretty self-explanatory (actually the comment made me a little more confused). After attempting to solve the problem independently myself, I came to the same code, so I think we can just remove the comment.

```suggestion
```

Reply:
Ok, I'll do that